src/components/Register.js

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import '../Register.css'
+import {register} from '../service/auth';
+import {Button, TextField} from "@material-ui/core";
+
+export class Register extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            login: '',
```

Comment:
It is not too critical, but this state looks a bit overloaded. You mixed up here different data like form data, patterns and error messages. At least form migth be moved into separate field 'form`